src/core/qgsmaplayer.cpp

```diff
@@ -1968,7 +1968,8 @@ bool QgsMapLayer::setDependencies( const QSet<QgsMapLayerDependency> &oDeps )
 void QgsMapLayer::setRefreshOnNotifyEnabled( bool enabled )
 {
   QgsDataProvider *lDataProvider = dataProvider();
-  if ( !lDataProvider )
+
+  if ( !lDataProvider || !lDataProvider->isValid() || !( providerType() == QStringLiteral("postgres") ) )
```

Comment:
please don't hardcode a provider name in here -- the better approach is to indicate in the provider's metadata that the provider can handle notifications, eg. following a similar approach to https://github.com/qgis/QGIS/pull/42154

Reply:
Actually it sounds very reasonable. But few lines below there are also hardcoded function ```connect( lDataProvider, &QgsVectorDataProvider::notify, ...```. In my opinion it also must be refactored. 
I don't think I have necessary skills and understanding of the QGIS code to do this. 
This PR makes this function ready for testing but not ready for production:)